contracts/payment/MCPayment.sol

```diff
@@ -231,6 +240,56 @@ contract MCPayment is Ownable2StepUpgradeable, EIP712Upgradeable {
         _transferERC20(paymentData, signer);
     }
 
+    /**
+     * @dev Withdraw in ERC-20 token utilizing permit (EIP-2612)
+     * @param permitSignature permit signature
+     * @param paymentData Payment data
+     * @param signature Signature of the payment data
+     */
+    function withdrawERC20Permit(
+        bytes memory permitSignature,
+        Iden3PaymentRailsERC20RequestV1 memory paymentData,
+        bytes memory signature
+    ) external {
+        if (_msgSender() != paymentData.recipient) {
+            revert PaymentError(
+                paymentData.recipient,
+                paymentData.nonce,
+                "MCPayment: invalid recipient"
+            );
+        }
+
+        address sponsor = _recoverERC20PaymentSignature(paymentData, signature);
```

Comment:
this is an address of issuer (or who creates payment-request), not signer of the Permit signature